src/main/java/it/pagopa/transactions/commands/handlers/v2/TransactionUpdateAuthorizationHandler.java

```diff
@@ -58,42 +60,143 @@ public Mono<BaseTransactionEvent<?>> handle(TransactionUpdateAuthorizationComman
-                        outcomeNpgGateway.getOperationId(),
-                        outcomeNpgGateway.getPaymentEndToEndId()
-                );
-            } else if (outcomeGateway instanceof OutcomeXpayGatewayDto
-                    || outcomeGateway instanceof OutcomeVposGatewayDto) {
-                authorizationData = new PgsTransactionGatewayAuthorizationData(
-                        authRequestDataExtracted.errorCode(),
-                        AuthorizationResultDto
-                                .fromValue(
-                                        authRequestDataExtracted.outcome()
-                                )
-                );
-            } else {
-                throw new IllegalStateException("Unexpected value: " + outcomeGateway);
-            }
+                                );
+                        default -> throw new InvalidRequestException("Unexpected value: " + outcomeGateway);
+                    };
 
-            return Mono.just(
-                    new it.pagopa.ecommerce.commons.documents.v2.TransactionAuthorizationCompletedEvent(
-                            transactionId.value(),
-                            new it.pagopa.ecommerce.commons.documents.v2.TransactionAuthorizationCompletedData(
-                                    authRequestDataExtracted.authorizationCode(),
-                                    authRequestDataExtracted.rrn(),
-                                    updateAuthorizationRequest.getTimestampOperation().toString(),
-                                    authorizationData
+            return isUpdateTransactionRequestValid(command)
+                    .thenReturn(
+                            new it.pagopa.ecommerce.commons.documents.v2.TransactionAuthorizationCompletedEvent(
+                                    transactionId.value(),
+                                    new it.pagopa.ecommerce.commons.documents.v2.TransactionAuthorizationCompletedData(
+                                            authRequestDataExtracted.authorizationCode(),
+                                            authRequestDataExtracted.rrn(),
+                                            updateAuthorizationRequest.getTimestampOperation().toString(),
+                                            authorizationData
+                                    )
                             )
                     )
-            )
                     .flatMap(transactionEventStoreRepository::save);
         } else {
             return alreadyProcessedError;
         }
 
     }
+
+    /**
+     * This method performs validation check against received update authorization
+     * request
+     *
+     * @param command the handler command
+     * @return an empty mono if request was validated successfully otherwise a mono
+     *         error with request validation failure details
+     */
+    private Mono<Void> isUpdateTransactionRequestValid(TransactionUpdateAuthorizationCommand command) {
```

Comment:
What about moving this validation logic inside `TransactionUpdateRequesetCommand`'s constructor instead?

Reply:
But here event store access is needed in order to reduce events and retrieve authorization data.
Also this is more of a semantic authorization request validation rather than a syntactic one (so a bean validation), performing checks also on authorization request timestamp and so on.
Idk, I would rather leave this validation logic here, or it can be moved to a utility class.
I see this piece of code as a pre-processing and maybe leave it here can help finding validation logic.
What do you think?

Reply:
Fixed with https://github.com/pagopa/pagopa-ecommerce-transactions-service/pull/412/commits/52d16a8accb30f8d01a0005a88b44eebe08546a4